core/src/main/scala/org/apache/spark/rdd/CartesianRDD.scala

```diff
@@ -53,11 +54,16 @@ class CartesianRDD[T: ClassTag, U: ClassTag](
   extends RDD[(T, U)](sc, Nil)
   with Serializable {
 
+  val numPartitionsInRdd1 = rdd1.partitions.length
```

Comment:
This change means that `rdd1.partitions` will be computed possibly more eagerly than it would have been computed before, which is potentially an internal behavior change w.r.t. the old behavior. In principle this should not matter, but if we ever wanted to consider backport of this change then I might weakly incline towards keeping the existing call pattern just for the sake of minimizing the scope of changes.

Reply:
@JoshRosen Thank you for your thoughtful advice, I will fix it.